spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/repository/RdbmsStreamDefinitionRepositoryTests.java

```diff
@@ -288,33 +289,106 @@ public void findAllSortTestASC() {
 	}
 
 	@Test
+	public void findAllPageableTestASC() {
+		Sort sort = new Sort(new Sort.Order(Sort.Direction.ASC, "DEFINITION_NAME"), new Sort.Order(Sort.Direction.ASC, "DEFINITION"));
+		Pageable pageable = new PageRequest(0, 10, sort);
+		String[] names = new String[]{"stream1", "stream2", "stream3"};
+		findAllPageable(pageable, names);
+	}
+
+	@Test
 	public void findAllSortTestDESC() {
 		Sort sort = new Sort(new Sort.Order(Sort.Direction.DESC, "DEFINITION_NAME"), new Sort.Order(Sort.Direction.DESC, "DEFINITION"));
 		String[] names = new String[]{"stream3", "stream2", "stream1"};
 		findAllSort(sort, names);
 	}
 
 	@Test
+	public void findAllPageableTestDESC() {
+		Sort sort = new Sort(new Sort.Order(Sort.Direction.DESC, "DEFINITION_NAME"), new Sort.Order(Sort.Direction.DESC, "DEFINITION"));
+		Pageable pageable = new PageRequest(0, 10, sort);
+
+		String[] names = new String[]{"stream3", "stream2", "stream1"};
+		findAllPageable(pageable, names);
+	}
+
+	@Test
 	public void findAllSortTestASCNameOnly() {
 		Sort sort = new Sort(new Sort.Order(Sort.Direction.ASC, "DEFINITION_NAME"));
 		String[] names = new String[]{"stream1", "stream2", "stream3"};
 		findAllSort(sort, names);
 	}
 
 	@Test
+	public void findAllPageableTestASCNameOnly() {
+		Sort sort = new Sort(new Sort.Order(Sort.Direction.ASC, "DEFINITION_NAME"));
+		Pageable pageable = new PageRequest(0, 10, sort);
+
+		String[] names = new String[]{"stream1", "stream2", "stream3"};
+		findAllPageable(pageable, names);
+	}
+
+	@Test
 	public void findAllSortTestASCDefinitionOnly() {
 		Sort sort = new Sort(new Sort.Order(Sort.Direction.DESC, "DEFINITION"));
 		String[] names = new String[]{"stream1", "stream2", "stream3"};
 		findAllSort(sort, names);
 	}
 
-	private void findAllSort(Sort sort, String[] expectedOrder) {
+	@Test
+	public void findAllPageableTestASCDefinitionOnly() {
+		Sort sort = new Sort(new Sort.Order(Sort.Direction.DESC, "DEFINITION"));
+		Pageable pageable = new PageRequest(0, 10, sort);
+
+		String[] names = new String[]{"stream1", "stream2", "stream3"};
+		findAllPageable(pageable, names);
+	}
+
+	@Test
+	public void findAllPageablePage2TestASC() {
+		Sort sort = new Sort(new Sort.Order(Sort.Direction.ASC, "DEFINITION_NAME"), new Sort.Order(Sort.Direction.ASC, "DEFINITION"));
+		Pageable pageable = new PageRequest(1, 2, sort);
+		String[] names = new String[]{"stream3"};
+		findAllPageable(pageable, names);
+	}
+
+	@Test
+	public void findAllPageablePage2TestDESC() {
+		Sort sort = new Sort(new Sort.Order(Sort.Direction.DESC, "DEFINITION_NAME"), new Sort.Order(Sort.Direction.DESC, "DEFINITION"));
+		Pageable pageable = new PageRequest(1, 2, sort);
+		String[] names = new String[]{"stream1"};
+		findAllPageable(pageable, names);
+	}
+
+	@Test
+	public void findAllPageableDefinitionStringTestDESC() {
+		Sort sort = new Sort(new Sort.Order(Sort.Direction.DESC, "DEFINITION"));
+		Pageable pageable = new PageRequest(1, 1, sort);
```

Comment:
Just curious why not set the PageRequest to 1,2 and check the second page.  for stream3.  For these tests stream 2 is always in the 2nd position.